fix(opentelemetry): Always use active span in `getInjectionData`

Lms24 opened the pull request:
WIP/Not sure what this change breaks. Let's see what CI has to say

Lms24 commented on dev-packages/node-integration-tests/suites/tracing/meta-tags-twp-errors/server.js:
```diff
@@ -0,0 +1,31 @@
+  Sentry.getClient().on('beforeEnvelope', envelope => {
+    const event = envelope[1][0][1];
+    if (event.exception.values[0].value === 'This is a test error') {
+      res.send({
+        traceData: Sentry.getTraceData(),
+        traceMetaTags: Sentry.getTraceMetaTags(),
+        errorTraceContext: event.contexts.trace,
+      });
```
I get this is not the way how we usually test event payloads but I don't think I can assert on a specific traceId in our runner `expectError` function which we don't know upfront.

Lms24 commented on dev-packages/node-integration-tests/suites/tracing/meta-tags-twp-errors/test.ts:
```diff
@@ -0,0 +1,43 @@
+      span_id: spanId,
+    });
+
+    expect(traceData).toEqual({
+      'sentry-trace': `${traceId}-${spanId}`,
+      baggage: expect.stringContaining(`sentry-trace_id=${traceId}`),
+    });
+
+    expect(traceMetaTags).toContain(`content="${traceId}-${spanId}"/>\n`);
+    expect(traceMetaTags).toContain(`sentry-trace_id=${traceId}`);
```
these trace and span ids were not the same as the error trace context's trace id without the change

Lms24 commented:
Hmm I don't understand why the node integration tests fail on CI. They pass locally so I suspect it's rather a test problem than a problem with the change. But I'll look into this on Monday.

mydea reviewed (approved):
I am not 100% sure anymore if/why this was necessary in the PR that introduced this (https://github.com/getsentry/sentry-javascript/pull/11564). But if tests pass, this should be fine I'd say 😅 

FWIW there is not _always_ a remote span active, only if an incoming trace is continued I believe.

Lms24 commented:
> FWIW there is not always a remote span active, only if an incoming trace is continued I believe.

Right... let me check the scenario for when there's no incoming trace then 😅

Lms24 commented:
> > FWIW there is not always a remote span active, only if an incoming trace is continued I believe.

> Right... let me check the scenario for when there's no incoming trace then 😅 

@mydea so the thing is, the test I added has no incoming trace headers. So looks like we _always_ have an active span now? 😅

mydea commented:
> > > FWIW there is not always a remote span active, only if an incoming trace is continued I believe.
> 
> > Right... let me check the scenario for when there's no incoming trace then 😅
> 
> @mydea so the thing is, the test I added has no incoming trace headers. So looks like we _always_ have an active span now? 😅

Not 100% sure, but I think we always have an active span if there is an incoming request. We probably (I think?) have none when outside of a request context 🤔 I think the propagator always creates a virtual span, but `extract` only runs for incoming requests!

Lms24 commented:
Ahh I see, so adding a test without a server should cover this scenario then, right?

mydea commented:
> Ahh I see, so adding a test without a server should cover this scenario then, right?

I'd say so 👍

Lms24 commented:
I added a test for non-server/request events. Also rewrote the existing test for events within a request. Looks like TwP mode now has consistent traces everywhere. Just gotta get them passing in CI now 😅
